include/reactor-cpp/assert.hh

```diff
@@ -8,22 +8,28 @@
 
 #pragma once
 
-// A custom definition of assert that mitigates unused variable warnings
-// when assertions are disabled
-#ifdef NDEBUG
-#define ASSERT(x)    \
-  do {               \
-    (void)sizeof(x); \
-  } while (0)
+#ifdef REACTOR_CPP_VALIDATE
+constexpr bool kDoRuntimeValidation = true;
 #else
-#include <cassert>
-#define ASSERT(x) assert(x)
+constexpr bool kDoRuntimeValidation = false;
 #endif
 
+// macro for silencing unused warnings my the compiler
+#define UNUSED(expr) do { (void)(expr); } while (0)
+
+#include <cassert>
 #include <sstream>
 #include <stdexcept>
 #include <string>
 
+constexpr inline void toggle_assert(bool condition) {
```

Comment:
I am a bit confused by this name. What does toggle mean in this context?

Also, this should probably become a member of the `reactor` namespace (same as `validate`)

Reply:
short hand for toggle-able assertion. Ok I will add it to the namespace.